chore(docs): add devnet.2 docs

critesjosh opened the pull request:
- adds docs build off of 3.0.0-devnet.2
- updates the links and version references in 2.0.2 docs to 2.0.4. it looks like a lot of changes, but its mostly just updating the urls to github source code that got hardcoded when the docs were build. I opted for updating the URLs in the existing v2.0.2 rather than rebuilding the docs for v2.0.4 because I know @sklppy88 has done a bunch of work on the v2.0.2 version of the docs
- adds `devnet` and `testnet` labels, so developers aren't looking at specific version numbers in the version dropdown
- makes v2.0.4 docs ("testnet" in the versions dropdown) resolve as docs.aztec.network/testnet/... in the url to future proof it somewhat
- updates "Getting started on Testnet" page to be "getting started on devnet for the devnet and future versions of the docs

alejoamiras reviewed (approved):
Couple of nits: 
- Devnet's  has a "Try Testnet", which also takes you to a page "try_testnet" with Testnet info. I was only able to get to Devnet's info going to "try testnet" and afterwards changing the version
- Migration notes on Devnet have [TBD]

Other than that I think it looks great. Amazing idea on using the "/testnet" and "/devnet" prefixes!

critesjosh commented:
thanks @alejoamiras. I updated the migration notes.

The "Try Testnet" pages have this info box to point people to the "Getting started on Devnet" page on the devnet version of the docs. 

<img width="1657" height="540" alt="image" src="https://github.com/user-attachments/assets/50f0d599-7954-43b3-a6de-dc27384c8a81" />

Since this is a navbar page, if it exists on 1 version of the docs, it needs to exist on all of the versions. I opted to keep it because it has been there, but I will check with Koen if we still want/need it around. Maybe we can get rid of it (in a future PR).

critesjosh committed:
chore(docs): add devnet.2 docs

- adds docs build off of 3.0.0-devnet.2
- updates the links and version references in 2.0.2 docs to 2.0.4. it looks like a lot of changes, but its mostly just updating the urls to github source code that got hardcoded when the docs were build. I opted for updating the URLs in the existing v2.0.2 rather than rebuilding the docs for v2.0.4 because I know @sklppy88 has done a bunch of work on the v2.0.2 version of the docs
- adds `devnet` and `testnet` labels, so developers aren't looking at specific version numbers in the version dropdown
- makes v2.0.4 docs ("testnet" in the versions dropdown) resolve as docs.aztec.network/testnet/... in the url to future proof it somewhat
- updates "Getting started on Testnet" page to be "getting started on devnet for the devnet and future versions of the docs
